src/simulators.rs

```diff
@@ -30,9 +27,8 @@ impl<G: Generator> Client<G> {
     // not. If a packet is generated, the last_gen time and till_next times are both updated.
     #[allow(dead_code)]
     fn next_packet(&mut self, time: u32) -> Option<Packet> {
```

Comment:
Same comment as above: `Client.next_packet(j); Client.next_packet(i);` where `i < j` is undefined behaviour. This is fine for the lab but it's along these lines why I was proposing a `.tick()` increment instead.